### PR TITLE
Feature/gcs 44 bob plugins

### PR DESCRIPTION
**Pull request by robertlong13**
Minimal set of features pulled over from ArgenTech's plugin.

### CODE REVIEW

File: MissionPlanner.sln [hunk 1]
```diff
@@ -891,6 +893,14 @@ Global
...
 		{0B99EBF2-B964-4AAC-9EC5-76353610AAAE}.Release|x86.ActiveCfg = Release|Any CPU
 		{0B99EBF2-B964-4AAC-9EC5-76353610AAAE}.Release|x86.Build.0 = Release|Any CPU
+		{B31E31DB-F30B-4B2E-9FD2-0E80CB10EC83}.Debug|Any CPU.ActiveCfg = Debug|Any CPU
+		{B31E31DB-F30B-4B2E-9FD2-0E80CB10EC83}.Debug|Any CPU.Build.0 = Debug|Any CPU
+		{B31E31DB-F30B-4B2E-9FD2-0E80CB10EC83}.Debug|x86.ActiveCfg = Debug|Any CPU
+		{B31E31DB-F30B-4B2E-9FD2-0E80CB10EC83}.Debug|x86.Build.0 = Debug|Any CPU
+		{B31E31DB-F30B-4B2E-9FD2-0E80CB10EC83}.Release|Any CPU.ActiveCfg = Release|Any CPU
+		{B31E31DB-F30B-4B2E-9FD2-0E80CB10EC83}.Release|Any CPU.Build.0 = Release|Any CPU
+		{B31E31DB-F30B-4B2E-9FD2-0E80CB10EC83}.Release|x86.ActiveCfg = Release|Any CPU
+		{B31E31DB-F30B-4B2E-9FD2-0E80CB10EC83}.Release|x86.Build.0 = Release|Any CPU
```
Comment (Pradeep-Carbonix) [inline MissionPlanner.sln:903]: Question - Are these 3 different from existing build configs?
`0E80CB10EC83}.Release|Any CPU.Build.0 = Release|Any CPU
		{B31E31DB-F30B-4B2E-9FD2-0E80CB10EC83}.Release|x86.ActiveCfg = Release|Any CPU
		{B31E31DB-F30B-4B2E-9FD2-0E80CB10EC83}.Release|x86.Build.0 = Release|Any CPU`

Comment (robertlong13) [reply MissionPlanner.sln:903]: These lines get automatically added by visual studio when you add a project to the solution. It's for this stuff, which I don't fully understand, but doesn't really matter.
image

File: plugins/Carbonix/UI/ActionsControl.cs [hunk 2]
```diff
@@ -0,0 +1,388 @@
...
+            // Get the number of waypoints in the currently loaded mission
+            int wpCount = Host.comPort.MAV.wps.Count - 1;
+            string wpno_str = Host.cs.wpno.ToString();
+
+            // Launch selection dialog and get resulting selected waypoint number
+            MissionPlanner.Controls.InputBox.Show("Enter waypoint number", "Set Waypoint", ref wpno_str);
+
+            if(!ushort.TryParse(wpno_str, out ushort wpno))
+            {
+                CustomMessageBox.Show("Invalid number format");
```
Comment (Pradeep-Carbonix) [inline plugins/Carbonix/UI/ActionsControl.cs:368]: Should this return ?

Comment (robertlong13) [reply plugins/Carbonix/UI/ActionsControl.cs:368]: Yes, great catch, will fix

File: plugins/Carbonix/UI/EmergencyTab.cs [hunk 3]
```diff
@@ -124,5 +124,52 @@ private void but_fenceenable_Click(object sender, EventArgs e)
...
+
+        private void arm_disarm(bool arm)
+        {
+            if (!Host.comPort.BaseStream.IsOpen)
+                return;
+
+            // arm the MAV
+            try
+            {
+                var action = arm ? "Arm" : "Disarm";
```
Comment (Pradeep-Carbonix) [inline plugins/Carbonix/UI/EmergencyTab.cs:145]: only one button is active (enabled) at any given time?
[but_arm.enabled == true if but_disarm.enabled == false]

File: plugins/Carbonix/UI/EmergencyTab.cs [hunk 4]
```diff
@@ -124,5 +124,52 @@ private void but_fenceenable_Click(object sender, EventArgs e)
             // Send aux function of 11:FENCE to high
             Host.comPort.doCommand(MAVLink.MAV_CMD.DO_AUX_FUNCTION, 11, 2, 0, 0, 0, 0, 0);
         }
+
+        private void but_arm_Click(object sender, EventArgs e)
```
Comment (Pradeep-Carbonix) [inline plugins/Carbonix/UI/EmergencyTab.cs:128]: suggestion - why not have a single button where text changes between ARM/DISARM (similar to CONNECT/DISCONNECT button)

Comment (robertlong13) [reply plugins/Carbonix/UI/EmergencyTab.cs:128]: Explicit Arm and Disarm buttons on this page are, in my opinion, safety features. It means that no matter what Mission Planner *thinks* the aircraft is doing, you always have a way to send a disarm packet, and always have a way to send an arm packet.

Comment (robertlong13) [reply plugins/Carbonix/UI/EmergencyTab.cs:128]: Would like @LachlanConn 's opinion on this too though.

File: plugins/Carbonix/UI/EmergencyTab.cs (hunk 3 above)
Comment (robertlong13) [reply plugins/Carbonix/UI/EmergencyTab.cs:145]: Both buttons should be active all the time (if I added enable/disable logic somewhere, please point it out and I'll fix). The `arm` variable here is a boolean input argument to the `arm_disarm` function, not tied to the enabled flag of a button.

File: plugins/Carbonix/UI/EmergencyTab.cs (hunk 4 above)
Comment (LachlanConn) [reply plugins/Carbonix/UI/EmergencyTab.cs:128]: I agree with Bob to have separate buttons especially on the emergency tab where simpler is better.

**Review by Pradeep-Carbonix — APPROVED** [review]

File: plugins/Carbonix/UI/ActionsControl.cs (hunk 2 above)
Comment (Pradeep-Carbonix) [inline plugins/Carbonix/UI/ActionsControl.cs:368]:
```suggestion
                CustomMessageBox.Show("Invalid number format");
                return;
```

**Pull request merged**